### PR TITLE
[tests-only] added scenarios for overwrite file with virus content in user share

### PR DESCRIPTION
**Pull request by KarunAtreya**
## Description
This PR adds the API test scenarios for overwrite file with virus content in user share:
- Scenario for trying to overwrite a file with the virus content in user share
- Scenario for trying to overwrite a file with the virus content in user share using spaces dav endpoint
## Related Issue
- Part of: https://github.com/owncloud/ocis/issues/6255

## Motivation and Context
- There was no test coverage for overwrite file with virus content in user share. So, this PR coves the required scenario test case.

## How Has This Been Tested?
- test environment:
- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [X] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [X] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>Core-API-Tests-ocis-storage-4</strong>](https://drone.owncloud.com/owncloud/ocis/23555/61/1) failed. Further test are cancelled...

File: tests/acceptance/features/apiAntivirus/antivirus.feature [hunk 1]
```diff
@@ -368,3 +368,39 @@ Feature: antivirus
...
       | new              |
       | spaces           |
+
+
+    Scenario Outline: try to overwrite a file with the virus content in user share
+      Given using <dav-path-version> DAV path
+      And user "Brian" has been created with default attributes and without skeleton files
+      And user "Alice" has created folder "uploadFolder"
+      And user "Alice" has shared folder "uploadFolder" with user "Brian" with permissions "all"
+      And user "Alice" has uploaded file with content "hello" to "uploadFolder/test.txt"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiAntivirus/antivirus.feature:378]: Can we check for both? A file in root as well as a file inside a folder, please adapt the rest of the scenario to overwrite both files and contents as well. Same for the scenario below
```suggestion
      And user "Alice" has created folder "uploadFolder"
      And user "Alice" has uploaded file with content "hello" to "uploadFolder/test.txt"
      And user "Alice" has uploaded file with content "hello" to "/test.txt"
      And user "Alice" has shared folder "uploadFolder" with user "Brian" with permissions "all"
      And user "Alice" has shared folder "/test.txt" with user "Brian" with permissions "all"
```

**Review by SagarGi — COMMENTED** [review]
other than above comments LGTM :+1:  Please rebase and reslove conflict.

**Review by SwikritiT — APPROVED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
Note: virus detection and notification is async. That is why the uploads get a `2xx` HTTP status - that just indicates that the uploaded stream-of-bytes has been received by the server. The virus check is then done by the server communicating with the antivirus scanning service. If a virus is detected, then the server takes action according to its policy (usually it deletes the uploaded file, keeping the previous content if there was any). The user has a notification waiting to read. The API test steps check that the notification is there "pending" on the server. (When testing a UI-based client like "web" we could test that the notification is displayed)

**Pull request merged**